static/js/tools/scatter/chart.tsx

```diff
@@ -165,26 +165,16 @@ function Chart(props: ChartPropsType): JSX.Element {
     }
   }
 
-  // Replot when data changes.
-  useEffect(() => {
-    if (props.display.chartType === ScatterChartType.MAP && !geoJsonFetched) {
-      loadSpinner(CONTAINER_ID);
-      return;
-    } else {
-      removeSpinner(CONTAINER_ID);
-    }
-    replot();
-  }, [props, geoJsonFetched]);
-
-  // Replot when chart width changes on sv widget toggle.
+  // Replot when props or chart width changes.
   useEffect(() => {
     const debouncedHandler = _.debounce(() => {
-      if (
-        props.display.chartType === ScatterChartType.SCATTER ||
-        geoJsonFetched
-      ) {
-        replot();
+      if (props.display.chartType === ScatterChartType.MAP && !geoJsonFetched) {
```

Comment:
this would also debounce updates on props/geojsonfetched. could you add todo to figure this out?

Reply:
found a way to only call the debounced method on window resize, updated!

Reply:
v cool!